internal/services/automanage/automanage_configuration_resource_test.go

```diff
@@ -0,0 +1,243 @@
+	r := AutoManageConfigurationProfileResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("antimalware.#").HasValue("1"),
+				check.That(data.ResourceName).Key("antimalware.0.exclusions.#").HasValue("1"),
+				check.That(data.ResourceName).Key("antimalware.0.exclusions.0.extensions").HasValue("exe"),
+				check.That(data.ResourceName).Key("antimalware.0.exclusions.0.processes").HasValue("svchost.exe"),
+				check.That(data.ResourceName).Key("antimalware.0.real_time_protection_enabled").HasValue("false"),
+				check.That(data.ResourceName).Key("antimalware.0.scheduled_scan_enabled").HasValue("true"),
+				check.That(data.ResourceName).Key("antimalware.0.scheduled_scan_day").HasValue("2"),
+				check.That(data.ResourceName).Key("antimalware.0.scheduled_scan_type").HasValue("Full"),
+				check.That(data.ResourceName).Key("antimalware.0.scheduled_scan_time_in_minutes").HasValue("1338"),
+				check.That(data.ResourceName).Key("automation_account_enabled").HasValue("false"),
+				check.That(data.ResourceName).Key("boot_diagnostics_enabled").HasValue("false"),
+				check.That(data.ResourceName).Key("defender_for_cloud_enabled").HasValue("false"),
+				check.That(data.ResourceName).Key("guest_configuration_enabled").HasValue("false"),
+				check.That(data.ResourceName).Key("status_change_alert_enabled").HasValue("false"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (r AutoManageConfigurationProfileResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := parse.AutomanageConfigurationID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	client := clients.Automanage.ConfigurationClient
+	resp, err := client.Get(ctx, id.ConfigurationProfileName, id.ResourceGroup)
+	if err != nil {
+		if response.WasNotFound(resp.Response.Response) {
+			return utils.Bool(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+	return utils.Bool(resp.Response.Response != nil), nil
+}
+
+func (r AutoManageConfigurationProfileResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctest-rg-%d"
+  location = "%s"
+}
+`, data.RandomInteger, data.Locations.Primary)
+}
+
+func (r AutoManageConfigurationProfileResource) basic(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+				%s
+
+resource "azurerm_automanage_configuration" "test" {
+  name                = "acctest-amcp-%d"
+  resource_group_name = azurerm_resource_group.test.name
+  location            = "%s"
+  antimalware {
```

Comment:
Basic test config should only contains properties that are Required but this block is marked as Optional so this should be removed?

Reply:
Agreed. Added test case antimalware and remove it configuration in basic.